### PR TITLE
Node.js: Add sample

### PR DESCRIPTION
**Pull request by mbana**
* Add preconfigured Cybersource JWT (`CONNECT_E_KEY`).
* Add sample in `nodejs` directory.

### CODE REVIEW

File: nodejs/Makefile [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	QUIET =
+	VERBOSE_FLAG = -v
+endif
+
+.DEFAULT_GOAL						:= all
+MAKEFLAGS 							+= --environment-overrides --warn-undefined-variables
+# .ONESHELL:
+SHELL										:= bash
+
+IMAGE_NAME							= connect-e-samples/nodejs
```
Comment (oliverreid) [inline nodejs/Makefile:16]: Is all this white space meant to be here?

File: nodejs/env [hunk 2]
```diff
@@ -0,0 +1,8 @@
+API_URL=e.test.connect.paymentsense.cloud
```
Comment (oliverreid) [inline nodejs/env:1]: Do these need to be passed from here and the docker-compose?

File: nodejs/src/access-token.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+          const error = new AccessTokenResponseError(
+            req.url,
+            response.status,
+            response.statusText,
+          );
+          throw error;
+        }
+        return response.json();
+      })
+      .then((data) => {
```
Comment (oliverreid) [inline nodejs/src/access-token.js:35]: Do we need the extra then? Could just log and throw in the status check if block

File: nodejs/src/access-token.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+//   "expiresAt": 1540457361
+// }
+class AccessTokenResponse {
+  constructor(id, expiresAt, orderId) {
+    this.id = id;
+    this.expiresAt = expiresAt;
+    this.orderId = orderId;
+  }
+
+  get paymentToken() {
```
Comment (oliverreid) [inline nodejs/src/access-token.js:115]: I've written more TypeScript recently, can we not just mark the id as public?

**Review by oliverreid — APPROVED** [review]

File: nodejs/Makefile (hunk 1 above)
Comment (mbana) [reply nodejs/Makefile:16]: I am not quite sure. I'll take another look.

`Makefile`s are indented using tabs so maybe that's issue.

If you checkout this branch and open the `Makefile` in the IDE, does it display correctly?

File: nodejs/src/access-token.js (hunk 3 above)
Comment (mbana) [reply nodejs/src/access-token.js:35]: `response.json()` returns a `Promise`. You cannot directly access the returned JSON in the first `then`, so I think it's needed.

File: nodejs/src/access-token.js (hunk 4 above)
Comment (mbana) [reply nodejs/src/access-token.js:115]: `id` is already public. JavaScript has no notion of public/private so ... I don't think that'd work.

This is effectively an alias to make the code using `AccessTokenResponse` to read.

File: nodejs/env (hunk 2 above)
Comment (mbana) [reply nodejs/env:1]: This file gets copied into the docker image.

Without it `make build-docker run-docker` would not work as in the `package.json` file there is `export $(cat env | xargs) && node src/index.js`.

Ideally, this file should be named `.env` and the lines in the `docker-compose.yml` should be:

```yaml
...
        CONNECT_E_KEY: ${CONNECT_E_KEY}
        API_URL: ${API_URL}
        WEB_URL: ${WEB_URL}
        PORT: ${PORT}
        GATEWAY_USERNAME: ${GATEWAY_USERNAME}
        GATEWAY_PASSWORD: ${GATEWAY_PASSWORD}
        MERCHANT_URL: ${MERCHANT_URL}
        WEBHOOK_URL: ${WEBHOOK_URL}
...
```

And `docker-compose up --build` will use the values defined in the `.env` file.

What do you think?

**Review by mbana — COMMENTED** [review]
Please re-review.

File: nodejs/Makefile (hunk 1 above)
Comment (oliverreid) [reply nodejs/Makefile:16]: Nope still not aligned. Other Makefiles look at on github eg https://gist.github.com/serinth/16391e360692f6a000e5a10382d1148c

File: nodejs/env (hunk 2 above)
Comment (oliverreid) [reply nodejs/env:1]: Yeah that sounds good. Better than having it in multiple places.

File: nodejs/src/access-token.js (hunk 3 above)
Comment (oliverreid) [reply nodejs/src/access-token.js:35]: gotcha.

**Review by oliverreid — APPROVED** [review]

File: nodejs/Makefile (hunk 1 above)
Comment (mbana) [reply nodejs/Makefile:16]: Thank you.
I will fix it.

File: nodejs/env (hunk 2 above)
Comment (mbana) [reply nodejs/env]: Thank you. It's resolved in last commit (https://github.com/Paymentsense/connect-e-samples/pull/2/commits/c8abdfb1bc8f3458992fb07e2648c3ce92aa64b6).

**Review by mbana — COMMENTED** [review]
Addressed code review points.

**Review by maxalencar — APPROVED** [review]

**Review by oliverreid — APPROVED** [review]

**Pull request merged**